vllm/distributed/device_communicators/pynccl.py

```diff
@@ -142,7 +142,7 @@ class ncclDataType_t(ctypes.c_int):
     ncclNumTypes = 10
 
     @classmethod
-    def from_torch(cls, dtype: torch.dtype) -> 'ncclDataType_t':
+    def from_torch(cls, dtype: torch.dtype) -> int:
```

Comment:
this is weird that mypy can't handle this, it should defenitely return the `c_int`

Reply:
Hmm when I added a breakpoint, it does return int. 

```
(Pdb) ncclDataType_t.from_torch(tensor.dtype)
7
(Pdb) type(ncclDataType_t.from_torch(tensor.dtype))
<class 'int'>
```

Do you think it is a bug?

Reply:
Well, technically, this function indeed returns `int`. `ctypes` will automatically convert int to `c_int` . Note that `ctypes.c_int` is a class/container to hold an `int`, and itself is not an `int`.

TL;DR; `def from_torch(cls, dtype: torch.dtype) -> int:` is correct.

Reply:
```
# enums
class ncclDataType_t(ctypes.c_int):
    pass


class ncclDataTypeEnum:
    ncclInt8 = ncclDataType_t(0)
    ncclChar = ncclDataType_t(0)
    ncclUint8 = ncclDataType_t(1)
    ncclInt32 = ncclDataType_t(2)
    ncclInt = ncclDataType_t(2)
    ncclUint32 = ncclDataType_t(3)
    ncclInt64 = ncclDataType_t(4)
    ncclUint64 = ncclDataType_t(5)
    ncclFloat16 = ncclDataType_t(6)
    ncclHalf = ncclDataType_t(6)
    ncclFloat32 = ncclDataType_t(7)
    ncclFloat = ncclDataType_t(7)
    ncclFloat64 = ncclDataType_t(8)
    ncclDouble = ncclDataType_t(8)
    ncclBfloat16 = ncclDataType_t(9)
    ncclNumTypes = ncclDataType_t(10)

    @classmethod
    def from_torch(cls, dtype: torch.dtype) -> "ncclDataType_t":
        if dtype == torch.int8:
            return cls.ncclInt8
        if dtype == torch.uint8:
            return cls.ncclUint8
        if dtype == torch.int32:
            return cls.ncclInt32
        if dtype == torch.int64:
            return cls.ncclInt64
        if dtype == torch.float16:
            return cls.ncclFloat16
        if dtype == torch.float32:
            return cls.ncclFloat32
        if dtype == torch.float64:
            return cls.ncclFloat64
        if dtype == torch.bfloat16:
            return cls.ncclBfloat16
        raise ValueError(f"Unsupported dtype: {dtype}")
```

I fixed this way so that it returns the correct type.

Reply:
oh @youkaichao I just saw your msg. So do you think ^ makes more sense or just return int? I can follow your suggestion

Reply:
I think technically inheriting c_int here is not necessary then? it is just like a simple enum

Reply:
yes, that's ChatGPT's fault 😉

Reply:
Fixed. PTAL...https://github.com/vllm-project/vllm/pull/4161/commits/81313dc5a1d32bc3cc2aed2a93304c6cfffedf39